File: wallet.js
class Wallet {
    constructor(balance = 0) {
        this.balance = balance;
    }

    deposit(amount) {
        this.balance += amount;
    }

    withdraw(amount) {
        if (amount > this.balance) {
            throw new Error("Insufficient funds");
        }
        this.balance -= amount;
    }

    get balance() {
        return this.balance;
    }
}

// Exemple d'utilisation
const wallet = new Wallet(100);
wallet.deposit(50);
console.log(wallet.balance); // 150
wallet.withdraw(25);
console.log(wallet.balance); // 125

File: wallet.cs
public class Wallet
{
    private int balance;

    public Wallet(int balance = 0)
    {
        this.balance = balance;
    }

    public void Deposit(int amount)
    {
        this.balance += amount;
    }

    public void Withdraw(int amount)
    {
        if (amount > this.balance)
        {
            throw new InsufficientFundsException();
        }
        this.balance -= amount;
    }

    public int GetBalance()
    {
        return this.balance;
    }
}

// Exemple d'utilisation
Wallet wallet = new Wallet(100);
wallet.Deposit(50);
Console.WriteLine(wallet.GetBalance()); // 150
wallet.Withdraw(25);
Console.WriteLine(wallet.GetBalance()); // 125
